modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java

```diff
@@ -167,10 +166,13 @@ public TemplateProcessor getTemplateProcessor() {
      * or other metadata files used by OpenAPI Generator.
      *
      * @param generateMetadata true: enable outputs, false: disable outputs
+     * @deprecated use {@link CodegenConfig#setGenerateMetadataFiles(boolean)} instead
+     * @throws NullPointerException if the config is not already instantiated
      */
     @SuppressWarnings("WeakerAccess")
+    @Deprecated
     public void setGenerateMetadata(Boolean generateMetadata) {
-        this.generateMetadata = generateMetadata;
+        this.config.setGenerateMetadataFiles(generateMetadata);
```

Comment:
Depending on reviewer preference, I could alternatively:
* default to `true` if `config == null`
* continue to maintain `generateMetadata` in `DefaultGenerator` and back out the commit that moves it into `CodegenConfig`, instead doing the `additionalProperties` check in `DefaultGenerator`
* delete this method instead of deprecating it
* something else!